### PR TITLE
Add documentation boilerplate

### PR DESCRIPTION
**Pull request by quitterie-lcs**
## Purpose 

Add documentation boilerplate.

## Proposal

Add `mkdocs` stack for documentation deployment automation

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -94,15 +102,16 @@ bootstrap: \
   create-api-test-db \
   migrate-api \
   migrate-app \
-  seed-lrs \
-  seed-xi
+  # seed-lrs \
+  # seed-xi
```
Comment (wilbrdt) [inline Makefile]:
```suggestion
   seed-lrs \
   seed-xi
```

File: .github/workflows/deploy-docs.yml [hunk 2]
```diff
@@ -0,0 +1,75 @@
+name: Documentation deployment CI
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
```
Comment (wilbrdt) [inline .github/workflows/deploy-docs.yml]: Here I think we would deploy the docs each time some activity occurs on a pull request, right?
What about having two separate actions:
- one that checks that the documentation builds correctly, when activity occurs on a pull_request
- one that build and deploy the docs on Github Pages, when pushed on main
WDYT?

File: Makefile [hunk 3]
```diff
@@ -47,6 +47,14 @@ WARREN_FRONTEND_IMAGE_TAG          ?= frontend-development
 WARREN_FRONTEND_IMAGE_BUILD_TARGET ?= development
 WARREN_FRONTEND_IMAGE_BUILD_PATH   ?= app/staticfiles/warren/assets/index.js
 
+# -- Documentation
+WARREN_DOCS_SERVER_PORT = 8000
+WARREN_DOCS_MIKE_PORT   = 8001
+WARREN_DOCS_ENV         = \
+													WARREN_DOCS_SERVER_PORT=$(WARREN_DOCS_SERVER_PORT) \
+													WARREN_DOCS_MIKE_PORT=$(WARREN_DOCS_MIKE_PORT)
```
Comment (wilbrdt) [inline Makefile]:
```suggestion
    WARREN_DOCS_SERVER_PORT=$(WARREN_DOCS_SERVER_PORT) \
    WARREN_DOCS_MIKE_PORT=$(WARREN_DOCS_MIKE_PORT)
```

File: Makefile [hunk 4]
```diff
@@ -307,6 +321,41 @@ test-api: \
...
+.PHONY: docs-openapi
+
+docs-build: ## build documentation site
+docs-build: \
+	docs-openapi
+	$(MKDOCS) build
+.PHONY: docs-build
+
+docs-deploy: ## build and deploy documentation site for all versions
+	@echo "Deploying docs with version main to gh-pages"
```
Comment (wilbrdt) [inline Makefile]: nit:
```suggestion
	@echo "Deploying docs with version dev to gh-pages"
```

File: bin/openapi [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+if [ ! -e "$file" ]; then
+    echo "File does not exist, creating..."
+    mkdir -p "$(dirname "$file")" || { echo "Error: Unable to create directory."; exit 1; }
+    touch "$file" || { echo "Error: Unable to create file."; exit 1; }
+    echo "{}" > "$file"  # Initializing with an empty JSON object
+    echo "File created successfully."
+else
+    echo "File already exists."
+    # Empty the file
+    > "$file"
```
Comment (wilbrdt) [inline bin/openapi:18]: Would it be a problem if it's not initialized with an empty JSON object?

File: docs/contribute.md [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+
+### 2. Clone the fork
+
+Clone the forked repository to your local machine and change the directory to
+the project folder using the following commands (replace `<your_fork>` with your
+GitHub username):
+
+```bash
+git clone https://github.com/<your_fork>/warren-tdbp.git
+cd warren
```
Comment (wilbrdt) [inline docs/contribute.md]:
```suggestion
cd warren-tdbp
```

File: docs/contribute.md [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+```bash
+git push origin feature/your-new-feature
+```
+
+### 5. Create a pull request
+
+To initiate a Pull Request (PR), head to Warren TdbP project's GitHub repository
+and click on <kbd>New Pull Request</kbd>.
+
+Set your branch as the source and Warren project's `main` branch as the target.
```
Comment (wilbrdt) [inline docs/contribute.md]:
```suggestion
Set your branch as the source and Warren TdbP project's `main` branch as the target.
```

File: docs/contribute.md [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+
+We use Git tags to trigger CI builds of the Warren TdbP's artifacts (PyPI/NPM
+packages and Docker images). To make a new release, depending on the service you
+are releasing, you need to apply the following Git tag pattern conventions:
+
+| Type         | Service  | Git tag pattern              | Example       |
+| ------------ | -------- | ---------------------------- | ------------- | 
+| Back-end     | API      | `v[0-9]+.[0-9]+.[0-9]+-api`  | `v2.0.1-api`  |
+|              | APP      | `v[0-9]+.[0-9]+.[0-9]+-app`  | `v1.3.6-app`  |
+| Front-end    | core     | `v[0-9]+.[0-9]+.[0-9]+-ui`   | `v5.1.0-ui`   |
```
Comment (wilbrdt) [inline docs/contribute.md]:
```suggestion
| Front-end    | tdbp     | `v[0-9]+.[0-9]+.[0-9]+-ui`   | `v5.1.0-ui`   |
```

File: bin/openapi (hunk 5 above)
Comment (quitterie-lcs) [reply bin/openapi:18]: The file is emptied each time the command is run to avoid appending the schema to existing content

**Review by wilbrdt — APPROVED** [review]

**Pull request merged**